src/main/java/io/zksync/sdk/musig/SchnorrMusigSigner.java

```diff
@@ -0,0 +1,183 @@
+package io.zksync.sdk.musig;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import io.zksync.sdk.musig.entity.AggregatedCommitment;
+import io.zksync.sdk.musig.entity.AggregatedPublicKey;
+import io.zksync.sdk.musig.entity.AggregatedSignature;
+import io.zksync.sdk.musig.entity.Commitment;
+import io.zksync.sdk.musig.entity.MusigSigner;
+import io.zksync.sdk.musig.entity.Precommitment;
+import io.zksync.sdk.musig.entity.Signature;
+import io.zksync.sdk.musig.exception.SchnorrMusigException;
+import io.zksync.sdk.musig.utils.Bytes;
+
+public class SchnorrMusigSigner {
+
+    private final SchnorrMusigNative musig;
+    private final MusigSigner signer;
+
+    SchnorrMusigSigner(SchnorrMusigNative musig, MusigSigner signer) {
+        this.musig = musig;
+        this.signer = signer;
+    }
+
+    public Signature sign(byte[] privateKey, byte[] message) throws SchnorrMusigException {
+        Signature.ByReference signature = new Signature.ByReference();
+
+        int code = this.musig.schnorr_musig_sign(this.signer.getPointer(), privateKey, privateKey.length, message,
+                message.length, signature);
+
+        SchnorrMusigResultCodes result = SchnorrMusigResultCodes.byCode(code);
+
+        if (result == SchnorrMusigResultCodes.OK) {
+            return signature;
+        } else {
+            throw new SchnorrMusigException(result);
+        }
+    }
+
+    public boolean verify(byte[] message, AggregatedSignature[] signatures,
```

Comment:
I believe we actually should pass a single AggregatedSignature, not an Array/Collection